api/src/main/java/org/eclipse/microprofile/config/Config.java

```diff
@@ -162,9 +169,10 @@
      * @param propertyName The configuration property name
      * @param propertyType The type into which the resolved property value should be converted
      * @return The resolved property value as an {@code Optional} wrapping the requested type
-     * @throws java.lang.IllegalArgumentException if the property cannot be converted to the specified type
+     *
+     * @throws IllegalArgumentException if the property cannot be converted to the specified type
      */
-    <T> Optional<T> getOptionalValue(String propertyName, Class<T> propertyType);
+    <T> Optional<T> getOptionalValue(String propertyName, Class<T> propertyType) throws IllegalArgumentException;
```

Comment:
I think we shouldn't declare the exceptions in the method signature. We were not doing it before and they are unchecked.

Reply:
removed.